README.md

```diff
@@ -1,28 +1,182 @@
 # Velveteen [![Build Status](https://travis-ci.com/thoughtbot/velveteen.svg?branch=master)](https://travis-ci.com/thoughtbot/velveteen)
 
-🚧 A work in-progress 🚧
-
 Transform your background jobs into a real data pipeline with Velveteen.
 
-## Install
+## We've got a pipeline problem
+
+Background jobs are an inevitable addition for many Rails applications. They
+typically start with a need to send email, then quickly grow into a collection
+of interdependent data processing jobs that are yearning to be a pipeline.
+Velveteen aims to insert itself near the beginning, providing support to build
+out the pipeline in a maintainable way.
+
+## 🚧 A work in-progress 🚧
+
+Prior to version 1.0, minor version updates are susceptible to breaking changes.
+
+## Getting started
+
+Velveteen requires Ruby `>= 2.5` and access to RabbitMQ.
 
 Add the following line to Gemfile:
 
 ```ruby
 gem "velveteen"
 ```
 
-and run `bundle install` from your shell.
+Run the bundle command to install it.
+
+Define a worker:
+
+```ruby
+# do_something.rb
+require "velveteen"
+
+Velveteen::Config.exchange_name = "velveteen_development"
+
+class DoSomething < Velveteen::Worker
+  self.routing_key = "something.do"
+
+  def perform
+    # do something
+  end
+end
+```
+
+Run it:
+
+```shell
+velveteen work do_something.rb DoSomething
+```
+
+Within the worker, you have access to the following:
+
+* `message` - A representation of the Rabbit message, which responds to:
+  * `body` - the raw message body
+  * `data` - the parsed JSON representation of the body
+  * `delivery_info` - the delivery info from Rabbit
+  * `headers` - the headers from Rabbit, from `properties.headers`
+  * `properties` - the properties Rabbit
+* `publish(payload, [options])` - Publishes the message to the exchange
+
+  The headers from the message the worker is consuming will be automatically
+  included when publishing new messages.
```

Comment:
I don't have a good suggestion, but I had to reread this to understand it.

Reply:
"Workers consume a message then publish a new one. When producing, a worker will automatically include the headers from the message it consumed."

I'm not fully clear on why that matters though. Is it about having all the headers from the start of the pipeline's execution?

Reply:
This is probably better suited to its own section in the readme, so the why can be more easily explained.

Reply:
Updated in fc92c4e0fa559500b9d61a10c465f31604f58526. Did that help?

Reply:
Definitely does for me. Thanks